vpr/src/route/route_timing.cpp

```diff
@@ -86,34 +65,37 @@ static int num_routing_failed = 0;
 
 /******************** Subroutines local to route_timing.cpp ********************/
 
+/** Attempt to route a single sink (target_pin) in a net.
+ * This returns a NetResultFlags, but the final flags are computed by timing_driven_route_net. */
 template<typename ConnectionRouter>
-static bool timing_driven_route_sink(ConnectionRouter& router,
-                                     const Netlist<>& net_list,
-                                     ParentNetId net_id,
-                                     unsigned itarget,
-                                     int target_pin,
-                                     const t_conn_cost_params cost_params,
-                                     const t_router_opts& router_opts,
-                                     RouteTree& tree,
-                                     std::vector<vtr::optional<const RouteTreeNode&>>& rt_node_of_sink,
-                                     SpatialRouteTreeLookup& spatial_rt_lookup,
-                                     RouterStats& router_stats,
-                                     route_budgets& budgeting_inf,
-                                     const RoutingPredictor& routing_predictor,
-                                     const std::vector<std::unordered_map<RRNodeId, int>>& choking_spots,
-                                     bool is_flat);
+static NetResultFlags timing_driven_route_sink(ConnectionRouter& router,
```

Comment:
Check that NetResultFlags has a good comment.
I think you should also beef this comment up some to describe assumptions in terms of data structs have been set up already for inputs, and what non-obvious inputs are (itarget, target_pin, ...)